### DISCUSSION TITLE
Is ZLS supposed to give error when re-assigning const?

### DISCUSSION THREAD

**Discussion by ziontee113** [zigtools/zls#1914] [Q&A] [answered]
Hi, with the following example:
```zig
    const y: u8 = 20;
    y = 44;
```
I can't see any diagnostics / errors about re-assigning the constant `y`.
Autocomplete / other diagnostics works fine for me.
Is this supported by ZLS? Or is there something wrong my my config?
Thank you very much.
I'm on Zig 0.13 and ZLS 0.13.

**Comment by techatrix** [comment, marked as answer]
This has been a long ongoing issue with ZLS because it does not perform semantic analysis of your code yet. We currently only report syntax errors and errors that reported by `zig ast-check`. There is a `enable_build_on_save` config option in ZLS that will run `zig build` on your project whenever you save a document and report back any compilation errors. Until Zig can provide incremental compilation, you may have to wait a bit until diagnostics show up depending on the size of your project.
